### PR TITLE
Add AWS kms support

### PR DESCRIPTION
**Pull request by yenteh-genies**
## Description
Add AWS KMS support to flow-go-sdk

______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-go-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Review by tarakby — COMMENTED** [review]
Thanks for adding the aws support 👌🏼  
I left minor comments and I'll try to test the signing functionality.

File: crypto/awskms/awskms_test.go
Comment (yenteh-genies) [reply crypto/awskms/awskms_test.go:75]: yes, these variable are needed to provide permission to access the AWS account having the key

File: crypto/awskms/awskms.go
Comment (yenteh-genies) [reply crypto/awskms/awskms.go]: Removed

File: crypto/awskms/awskms.go
Comment (yenteh-genies) [reply crypto/awskms/awskms.go]: addressed

File: crypto/awskms/awskms_test.go
Comment (yenteh-genies) [reply crypto/awskms/awskms_test.go]: removed

File: crypto/awskms/awskms.go
Comment (sideninja) [inline crypto/awskms/awskms.go:48]: No need to export type and its' functions.

File: crypto/awskms/awskms.go
Comment (sideninja) [inline crypto/awskms/awskms.go:66]: No need to export

File: crypto/util.go
Comment (sideninja) [inline crypto/util.go:11]: Should this be part of kms package instead? is it used anywhere else?

**Review by sideninja — COMMENTED** [review]
Looks good, left some comments. I will leave the final say to @tarakby

File: crypto/awskms/awskms.go
Comment (yenteh-genies) [reply crypto/awskms/awskms.go:48]: Key needs to be exported to be used to create the key. I follow the existing pattern for the cloudkms. Will lowercase some other functions that are not required to be exported

**Comment by sideninja** [comment]
Related https://github.com/onflow/flow-cli/issues/436

File: crypto/awskms/awskms.go
Comment (tarakby) [reply crypto/awskms/awskms.go:66]: I think this function is needed still. The [test](https://github.com/geniesinc/flow-go-sdk/blob/aws-kms-support/crypto/awskms/awskms_test.go#L56) how it can be used outside the package.

File: crypto/awskms/signer.go
Comment (tarakby) [reply crypto/awskms/signer.go]: Thanks for adding `crypto/kms_internal`. I think if the the sub-package is called `internal`, the capital methods (here `ParseSignature `) won't be exported outside `crypto/`.  But I'm not sure if that remains true if the package only has "internal" suffix. Can you confirm that's the case? otherwise we could just use "internal"

File: crypto/util.go
Comment (tarakby) [reply crypto/util.go]: added a related comment here https://github.com/onflow/flow-go-sdk/pull/340#discussion_r1101941778

**Comment by tarakby** [comment]
The PR is clean, looks good to me, but I would like to test before merging. I already reached out to IT to get an aws account to test and that's still pending. I'll merge once that's done.

**Comment by yenteh-genies** [comment]
> The PR is clean, looks good to me, but I would like to test before merging. I already reached out to IT to get an aws account to test and that's still pending. I'll merge once that's done.

Thank you

**Comment by tarakby** [comment]
@yenteh-genies I was able to test the PR code today. I made some changes, one of them was necessary for the test to work (the region config). I added changes to this PR https://github.com/geniesinc/flow-go-sdk/pull/2. 

Feel free to review and merge them into your branch, and then we can merge this current PR. Thanks

**Comment by yenteh-genies** [comment]
> @yenteh-genies ping ^

Approved the PR suggestion and merged

**Review by tarakby — APPROVED** [review]

**Comment by tarakby** [comment]
@sideninja I think I don't have permissions to merge the PR

**Review by sideninja — APPROVED** [review]

**Pull request merged**
